orttraining/orttraining/core/optimizer/gist_encode_decode.h

```diff
@@ -34,14 +33,16 @@ class GistEncodeDecode : public RewriteRule {
       {"Relu", {"ReluGrad", "Shape", "Reshape"}}};
 
   GistEncodeDecode() noexcept : RewriteRule("GistEncodeDecode") {}
-  GistEncodeDecode(int op_type, std::string compr_type) noexcept : RewriteRule("GistEncodeDecode"), operator_type(op_type), compression_type(std::move(compr_type)) {}
+  GistEncodeDecode(int op_type, std::string compr_type) noexcept : RewriteRule("GistEncodeDecode"), operator_type(op_type), compression_type_(std::move(compr_type)) {}
 
  private:
   int GenerateDecodePriority() const { return priority_generator_--; };
   std::vector<std::string> TargetOpTypes() const noexcept override;
   bool SatisfyCondition(const Graph& graph, const Node& node, const logging::Logger& logger) const override;
   Status Apply(Graph& graph, Node& node, RewriteRuleEffect& rule_effect, const logging::Logger& logger) const override;
   bool AddEncodeDecode(Graph& graph, Node& curr_node, std::string compression_type, const logging::Logger& logger) const;
+
+  std::string compression_type_;
```

Comment:
while we're updating this, can compression_type_ be const?

Reply:
Good idea! Done.